Model/Sync/Catalog/Processor.php

```diff
@@ -214,205 +214,174 @@ public function process($forceSyncProducts = [], $storeIds = [])
-                    }
+            $response = $this->processRequest($apiRequestParams, $yotpoFormatItemData);
+
+            $yotpoIdKey = $isVisibleVariantsSync ? 'visible_variant_yotpo_id' : 'yotpo_id';
+            $yotpoIdValue = $apiRequestParams['yotpo_id'] ?: 0;
+            $responseCode = $response->getData('status');
+            $syncDataRecordToUpdate = $this->prepareSyncTableDataToUpdate($itemEntityId, $yotpoIdKey, $yotpoIdValue, $storeId, $responseCode);
+            if (!$isVisibleVariantsSync) {
+                $syncDataRecordToUpdate['yotpo_id_parent'] = $apiRequestParams['yotpo_id_parent'] ?: 0;
+            }
+            if ($this->coreConfig->canUpdateCustomAttributeForProducts($syncDataRecordToUpdate['response_code'])) {
+                if ($this->isSyncingAsMainEntity()) {
+                    $this->updateProductSyncAttribute($attributeDataToUpdate);
                 }
+            }
 
-                $returnResponse = $this->processResponse(
-                    $apiParam,
-                    $response,
-                    $tempSqlArray,
-                    $itemData,
-                    $externalIds,
-                    $visibleVariants
-                );
+            $returnResponse = $this->processResponse(
+                $apiRequestParams,
+                $response,
+                $syncDataRecordToUpdate,
+                $yotpoFormatItemData,
+                $externalIdsWithConflictResponse,
+                $isVisibleVariantsSync
+            );
 
-                $tempSqlArray = $returnResponse['temp_sql'];
-                $externalIds = $returnResponse['external_id'];
+            $processedSyncDataRecordToUpdate = $returnResponse['temp_sql'];
+            $externalIdsWithConflictResponse = $returnResponse['external_id'];
 
-                if (isset($this->retryItems[$storeId][$itemId])) {
-                    unset($this->retryItems[$storeId][$itemId]);
-                }
+            if (isset($this->retryItems[$storeId][$itemEntityId])) {
+                unset($this->retryItems[$storeId][$itemEntityId]);
+            }
 
-                if (count($returnResponse['four_not_four_data'])) {
-                    foreach ($returnResponse['four_not_four_data'] as $retryId) {
-                        if ($this->isImmediateRetry($response, $this->entity, $visibleVariants.$retryId, $storeId)) {
-                            $this->setImmediateRetryAlreadyDone($this->entity, $visibleVariants.$retryId, $storeId);
-                            $this->retryItems[$storeId][$retryId] = $retryId;
-                        }
+            if (count($returnResponse['four_not_four_data'])) {
+                foreach ($returnResponse['four_not_four_data'] as $retryId) {
+                    if ($this->isImmediateRetry($response, $this->entity, $isVisibleVariantsSync.$retryId, $storeId)) {
+                        $this->setImmediateRetryAlreadyDone($this->entity, $isVisibleVariantsSync.$retryId, $storeId);
```

Comment:
@EranAvidor ,

This line was removed at https://github.com/YotpoLtd/magento2-module-core/pull/128/commits/3a0ad98b4a7061f654467a8575f1f125b74a39af

Reply:
Thanks. This line is not included on the `manaf-product-sync-404-issue` branch.
Please see [this](https://github.com/YotpoLtd/magento2-module-core/pull/128/files#diff-0e872f4c5096af743b1505e9f25f2b9b11b89c98b9e7df4587bbfbc6ac08134cR312-R313).